src/vue/pages/BusinessViewer.vue

```diff
@@ -0,0 +1,270 @@
+
+        <div class="business-viewer__shop">
+          <h1 ref="shop" class="business-viewer__title">
+            {{ 'business-viewer.shop' | globalize }}
+          </h1>
+        </div>
+        <atomic-swaps-explore
+          :business-id="business.accountId"
+        />
+      </template>
+      <template v-else-if="!isLoaded && !isFailed">
+        <loader
+          :message-id="'business-viewer.loading-msg'"
+        />
+      </template>
+      <template v-if="isFailed">
+        <no-data-message
+          icon-name="castle"
+          :title="'business-viewer.error-title' | globalize"
+          :message="'business-viewer.error-msg' | globalize"
+        />
+      </template>
+    </div>
+  </div>
+</template>
+
+<script>
+import BusinessViewerDescription from '@/vue/pages/business-viewer/BusinessViewerDescription'
+import AtomicSwapsExplore from '@/vue/pages/atomic-swaps/AtomicSwapsExplore'
+import NoDataMessage from '@/vue/common/NoDataMessage'
+import Loader from '@/vue/common/Loader'
+
+import { vueRoutes } from '@/vue-router/routes'
+import { api } from '@/api'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import { BusinessRecord } from '@/js/records/entities/business.record'
+import { Bus } from '@/js/helpers/event-bus'
+import { mapGetters, mapActions, mapMutations } from 'vuex'
+import { vuexTypes } from '@/vuex'
+
+export default {
+  name: 'business-viewer',
+
+  components: {
+    BusinessViewerDescription,
+    AtomicSwapsExplore,
+    Loader,
+    NoDataMessage,
+  },
+
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+  },
+
+  data () {
+    return {
+      vueRoutes,
+      business: {},
+      isLoaded: false,
+      isFailed: false,
+      isSubmitting: false,
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      accountId: vuexTypes.accountId,
+      myBusinesses: vuexTypes.myBusinesses,
+      isBusinessToBrowse: vuexTypes.isBusinessToBrowse,
+      businessToBrowse: vuexTypes.businessToBrowse,
+    }),
+
+    isMyBusiness () {
+      return Boolean(this.myBusinesses.find(business => {
+        return business.accountId === this.business.accountId
+      })
+      )
+    },
+  },
+
+  async created () {
+    if (this.isBusinessToBrowse) {
+      this.business = this.businessToBrowse
+    } else {
+      await this.getBusiness()
+    }
+    await this.loadMyBusinesses()
+    this.setBusinessStatsQuoteAsset(this.business.statsQuoteAsset)
+    this.isLoaded = true
+  },
+
+  beforeDestroy () {
+    this.clearBusinessToBrowse()
+  },
+
+  methods: {
+    ...mapActions({
+      loadMyBusinesses: vuexTypes.LOAD_MY_BUSINESSES,
+    }),
+
+    ...mapMutations({
+      setBusinessStatsQuoteAsset: vuexTypes.SET_BUSINESS_STATS_QUOTE_ASSET,
+      clearBusinessToBrowse: vuexTypes.CLEAR_BUSINESS_TO_BROWSE,
+    }),
+
+    async getBusiness () {
+      try {
+        const endpoint = `/integrations/dns/businesses/${this.id}`
+        const { data } = await api.get(endpoint)
```

Comment:
const { data } = await api.get(`/integrations/dns/businesses/${this.id}`)

Reply:
changed